Source/Scene/Camera.js

```diff
@@ -361,6 +362,22 @@ Camera.prototype._updateCameraChanged = function () {
 
   var percentageChanged = camera.percentageChanged;
 
+  var currentHeading = camera.heading;
+
+  if (!defined(camera._changedHeading)) {
+    camera._changedHeading = currentHeading;
+  }
+
+  var headingChangedPercentage =
+    (Math.abs(camera._changedHeading - currentHeading) /
+      ((camera._changedHeading + currentHeading) / 2)) *
+    100;
```

Comment:
It is based on the relative change formula. Two definitions are stated on the Wikipedia page.
- https://en.wikipedia.org/wiki/Relative_change_and_difference

This formula is also used in various online calculators.
- https://www.calculatorsoup.com/calculators/algebra/percent-difference-calculator.php

Does this seem okay to you? Is there any other formula that I should be using?

Reply:
There are a couple issues with it:

* If `_changedHeading` and `currentHeading` are both zero it will result in a divide by zero
* It will return drastically different results for the same heading delta. For example if the heading goes between 0 and 1 degrees it will return 200 but if the heading goes between 179 and 180 degrees it will return 0.5571030640668461. I would expect the results to be exactly the same.

Looking at how the direction percentage is calculated it finds the angle between the two vectors and divides that by the camera field of view. There isn't really an equivalent camera property that heading could be expressed as a percentage of, so I think a simple `abs(changedHeading - currentHeading)/2PI)` is the best approach here, though you'd have to rotate the camera by 180 degrees to trigger the event since `percentageChanged` is 0.5 by default, which seems like a lot. Maybe it's ok to fudge the equation a bit and divide by PI or PI/2.

Reply:
Do you mean
`abs((changedHeading - currentHeading)/2)PI)`
or
`abs(changedHeading - currentHeading)/(2PI))`?

Regardless, I think the following makes the most sense:
`abs(changedHeading - currentHeading)/(PI))`

Reply:
To be exact, I think it would like like this in the code:
`var headingChangedPercentage = Math.abs(camera._changedHeading - currentHeading)/Math.PI;`

Reply:
Looks good 👍

Reply:
Sorry one more thing to watch out for: does the heading ever wrap around from 360 to 0 or does it keep increasing? If this jump exists it will need to be factored into the equation.

Reply:
From my testing, the heading seems to go from 0 to about 6.251332195769063 and then back to around 0.

Reply:
The conclusion from our offline discussion:   For any movement that has a delta of over 358 degrees, we will assume that it is a one-degree change.